File: app/src/main/java/com/example/merchio/adapters/AdminUserAdapter.java
package com.example.merchio.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.merchio.R;
import com.example.merchio.models.AdminUser;

import java.util.ArrayList;
import java.util.List;

public class AdminUserAdapter extends RecyclerView.Adapter<AdminUserAdapter.UserViewHolder> {

    public interface OnUserActionListener {
        void onEdit(AdminUser user);
    }

    private final List<AdminUser> users = new ArrayList<>();
    private final OnUserActionListener listener;

    public AdminUserAdapter(OnUserActionListener listener) {
        this.listener = listener;
    }

    public void setUsers(List<AdminUser> newUsers) {
        users.clear();
        users.addAll(newUsers);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public UserViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_admin_user, parent, false);
        return new UserViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull UserViewHolder holder, int position) {
        AdminUser user = users.get(position);

        holder.tvNumber.setText(String.valueOf(position + 1));
        holder.tvName.setText(empty(user.getName()) ? "Merchio User" : user.getName());
        holder.tvStatus.setText(user.isActive() ? "Status : Active" : "Status : Inactive");
        holder.tvStatus.setAlpha(user.isActive() ? 1f : 0.65f);

        holder.btnEdit.setOnClickListener(v -> {
            if (listener != null) {
                listener.onEdit(user);
            }
        });
    }

    @Override
    public int getItemCount() {
        return users.size();
    }

    private boolean empty(String value) {
        return value == null || value.trim().isEmpty();
    }

    static class UserViewHolder extends RecyclerView.ViewHolder {
        TextView tvNumber, tvName, tvStatus, btnEdit;

        UserViewHolder(@NonNull View itemView) {
            super(itemView);
            tvNumber = itemView.findViewById(R.id.tv_user_number);
            tvName = itemView.findViewById(R.id.tv_user_name);
            tvStatus = itemView.findViewById(R.id.tv_user_status);
            btnEdit = itemView.findViewById(R.id.btn_edit_user);
        }
    }
}

File: app/src/main/java/com/example/merchio/adapters/AdminUserAdapter.kt
package com.example.merchio.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.merchio.R
import com.example.merchio.data.model.ProfileDto

class AdminUserAdapter(private val onEdit: (ProfileDto) -> Unit) : RecyclerView.Adapter<AdminUserAdapter.AdminUserViewHolder>() {
    private val items = mutableListOf<ProfileDto>()
    fun submitList(data: List<ProfileDto>) { items.clear(); items.addAll(data); notifyDataSetChanged() }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AdminUserViewHolder =
        AdminUserViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_admin_user, parent, false))
    override fun onBindViewHolder(holder: AdminUserViewHolder, position: Int) = holder.bind(items[position], position)
    override fun getItemCount(): Int = items.size

    inner class AdminUserViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val number: TextView = itemView.findViewById(R.id.tv_user_number)
        private val name: TextView = itemView.findViewById(R.id.tv_user_name)
        private val status: TextView = itemView.findViewById(R.id.tv_user_status)
        private val edit: View = itemView.findViewById(R.id.btn_edit_user)
        fun bind(item: ProfileDto, pos: Int) {
            number.text = "${pos + 1}."
            name.text = "${item.displayName}\n${item.email ?: "-"}"
            status.text = if (item.isActive) item.role.uppercase() else "INACTIVE"
            edit.setOnClickListener { onEdit(item) }
        }
    }
}
